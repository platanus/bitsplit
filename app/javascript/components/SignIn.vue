<template>
  <div>
    <p>Route: {{ routeName }}</p>
    <form @submit.prevent="handleSubmit">
      <div>
        <label for="email">email</label>
        <input type="text" v-model="email" name="email" />
      </div>
      <div>
        <label htmlFor="password">Password</label>
        <input type="password" v-model="password" name="password" />
      </div>
      <div>
        <button :disabled="loading">
          Login
        </button>
      </div>
    </form>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'

export default {
  name: 'SignIn',
  data() {
    return {
      routeName: 'Sign In',
      email: '',
      password: '',
      loading: false
    }
  },
  computed: {
    ...mapState('user', ['currentUser'])
  },
  methods: {
    ...mapActions('user', ['signIn']),
    handleSubmit(e) {
      const { email, password } = this
      if (email && password) {
        this.signIn({ email, password })
          .then(() => {
            console.log('success')
            this.$router.push('/home')
          })
          .catch(() => {
            console.log('error')
          })
      }
    }
  }
}
</script>

<style lang="scss"></style>
