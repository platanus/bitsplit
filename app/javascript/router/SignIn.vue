<template>
  <div class="items-center h-screen w-full bg-primary p-12 mt-24">
    <center>
      <div class="w-full max-w-xs ">
        <img
          :src="bitsplitLogo"
          class="mb-10"
          align="top"
        >
        <form
          @submit.prevent="handleSubmit"
          class="bg-primary rounded pt-6 pb-8 mb-4 "
        >
          <div class="mb-4">
            <textInput
              field-id="email"
              field-type="text"
              field-placeholder="Correo"
              field-name="email"
              v-model="email"
            />
          </div>
          <div>
            <passwordInput
              field-id="password"
              field-name="password"
              field-placeholder="ContraseÃ±a"
              v-model="password"
            />
          </div>
          <submitButton :loading="signInLoading">
            Ingresar
          </submitButton>
          <p class="txt-field items-center mt-10 underline ">
            <router-link :to="signUpRoute">
              Crear una cuenta
            </router-link>
          </p>
        </form>
      </div>
    </center>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex';
import textInput from '../components/Input';
import passwordInput from '../components/PasswordInput';
import submitButton from '../components/SubmitButton';
import logo from '../assets/bitsplit-logo.svg';

export default {
  name: 'SignIn',
  data() {
    return {
      routeName: 'Sign In',
      email: '',
      password: '',
      signUpRoute: 'sign-up',
      bitsplitLogo: logo,
    };
  },
  components: {
    textInput,
    passwordInput,
    submitButton,
  },
  computed: {
    ...mapState('user', ['currentUser', 'signInLoading']),
  },
  methods: {
    ...mapActions('user', ['signIn']),
    handleSubmit() {
      const { email, password } = this;
      if (email && password) {
        this.signIn({ email, password })
          .then(() => {
            // TODO logger
          })
          .catch(() => {
            console.error('error');
          });
      }
    },
  },
};
</script>
