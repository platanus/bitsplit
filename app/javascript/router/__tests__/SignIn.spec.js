import { shallowMount } from '@vue/test-utils'
import SignIn from '../SignIn'

describe('SignIn', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(SignIn)
    expect(wrapper).toMatchSnapshot()
  })
})
