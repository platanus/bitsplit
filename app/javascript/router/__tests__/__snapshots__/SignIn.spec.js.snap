// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn matches snapshot 1`] = `
<div id="SignInForm">
  <center>
    <div class="w-full max-w-xs">
      <form class="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4">
        <div class="mb-4">
          <inputlabel-stub fieldname="Email" fieldfor="email"></inputlabel-stub>
          <textinput-stub value="" fieldid="email" fieldtype="text" fieldplaceholder="Email" fieldname="email" classmod=""></textinput-stub>
        </div>
        <div class="mb-4">
          <inputlabel-stub fieldname="Password" fieldfor="password"></inputlabel-stub>
          <passwordinput-stub value="" fieldid="password" fieldname="password"></passwordinput-stub>
          <submitbutton-stub fieldplaceholder="Log In" classmod="bg-blue-500 hover:bg-blue-700"></submitbutton-stub>
        </div>
      </form>
    </div>
  </center>
</div>
`;
